### PR TITLE
Create combination_sum.py

### PR DESCRIPTION
**Pull request by creatornadiran**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by creatornadiran** [comment]
@cclauss @CaedenPH I have reopened pull request for combination_sum.py could you review my commit again?

File: backtracking/combination_sum.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        answer.append(path.copy())
+    else:
+        for index in range(previous_index, len(candidates)):
+            if target >= candidates[index]:
+                path.append(candidates[index])
+                backtrack(candidates, path, answer, target - candidates[index], index)
+                path.pop(len(path) - 1)
+
+
+def combination_sum(candidates: list, target: int) -> list:
```
Comment (cclauss) [inline backtracking/combination_sum.py:34]: Let's add some [doctests](https://docs.python.org/3/library/doctest.html)...
```suggestion
def combination_sum(candidates: list, target: int) -> list:
    """
    >>> combination_sum([2, 3, 5], 8)
    [[2, 2, 2, 2], [2, 3, 3], [3, 5]]
    >>> combination_sum([2, 3, 6, 7], 7)
    [[2, 2, 3], [7]]
    >>> combination_sum([-8, 2.3, 0], 1)
    Put the right answer here
    """
```

File: backtracking/combination_sum.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    """
+    >>> combination_sum([2, 3, 5], 8)
+    [[2, 2, 2, 2], [2, 3, 3], [3, 5]]
+    >>> combination_sum([2, 3, 6, 7], 7)
+    [[2, 2, 3], [7]]
+    >>> combination_sum([-8, 2.3, 0], 1)
+    Put the right answer here
```
Comment (cclauss) [inline backtracking/combination_sum.py:41]: Please fix this line so that the tests are green ✅

Comment (creatornadiran) [reply backtracking/combination_sum.py]: Thanks, I didn't notice. I have added constraints instead.

File: backtracking/combination_sum.py [hunk 3]
```diff
@@ -37,8 +44,6 @@ def combination_sum(candidates: list, target: int) -> list:
     [[2, 2, 2, 2], [2, 3, 3], [3, 5]]
     >>> combination_sum([2, 3, 6, 7], 7)
     [[2, 2, 3], [7]]
-    >>> combination_sum([-8, 2.3, 0], 1)
```
Comment (cclauss) [inline backtracking/combination_sum.py]: Why remove this test?

Comment (creatornadiran) [reply backtracking/combination_sum.py]: Because it breaks the constraints and gives "recursion depth exceeded" error. Should i still add?

Comment (creatornadiran) [reply backtracking/combination_sum.py]: It was breaking the constraints. I have added again.

**Review by cclauss — APPROVED** [review]
Rock and roll!

**Comment by creatornadiran** [comment]
Thanks for help.

**Pull request merged**